lib/allocation.c

```diff
@@ -216,8 +216,8 @@ get_unused_cos(const unsigned id,
                         if (m_cpu->cores[i].l2_id != id)
                                 continue;
                 } else {
-                        /* L2 not requested so looking at socket scope */
-                        if (m_cpu->cores[i].socket != id)
+			/* L2 not requested so looking at l3cat_id scope */
```

Comment:
What about mba_id?

Reply:
This function is only called from l2 or l3 cat id context.

Reply:
This is used by hw_alloc_assign function that can be used for MBA too

Reply:
ok. Yes. fixed it now. Thanks